RKNotificationHubTests/RKNotificationHubTests.m

```diff
@@ -1,40 +1,277 @@
-//
+#import <Specta/Specta.h>
+#import <Expecta/Expecta.h>
+#import <FBSnapshotTestCase/FBSnapshotTestCase.h>
+#import <Expecta+Snapshots/EXPMatchers+FBSnapshotTest.h>
 
-#import <UIKit/UIKit.h>
-#import <XCTest/XCTest.h>
+#import "RKNotificationHub.h"
 
-@interface RKNotificationHubTests : XCTestCase
+@implementation UIView(SpecFlywheel)
+
++ (void)animateWithDuration:(NSTimeInterval)duration
+                 animations:(void (^)(void))animations
+                 completion:(void (^)(BOOL finished))completion {
+    if (animations)
+        animations();
+    if (completion)
+        completion(YES);
+}
 
 @end
 
-@implementation RKNotificationHubTests
+SpecBegin(RKNotificationHub)
 
-- (void)setUp {
-    [super setUp];
-    // Put setup code here. This method is called before the invocation of each test method in the class.
-}
+// setup
 
-- (void)tearDown {
-    // Put teardown code here. This method is called after the invocation of each test method in the class.
-    [super tearDown];
-}
+describe(@"initWithView:", ^{
+    __block UIView *view;
+    __block RKNotificationHub *subject;
+    
+    beforeAll(^{
+        view = [[UIView alloc] initWithFrame:CGRectZero];
+        subject = [[RKNotificationHub alloc] initWithView:view];
+    });
+    
+    it(@"should initialize hub with given view as hubView", ^{
+        expect(subject.hubView).to.equal(view);
+    });
+    
+    it(@"should initialize hub with count 0", ^{
+        expect([subject count]).to.equal(0);
+    });
+});
 
-- (void)testExample {
-    // This is an example of a functional test case.
-    XCTAssert(YES, @"Pass");
-}
+describe(@"setView:andCount:", ^{
+    __block UIView *view;
+    __block RKNotificationHub *subject;
+    
+    beforeAll(^{
+        view = [[UIView alloc] initWithFrame:CGRectZero];
+        subject = [[RKNotificationHub alloc] initWithView:nil];
+        [subject setView:view andCount:2];
+    });
+    
+    it(@"should set hub view", ^{
+        expect(subject.hubView).to.equal(view);
+    });
+    
+    it(@"should set count", ^{
+        expect([subject count]).to.equal(2);
+    });
+});
 
-- (void)testPerformanceExample {
-    // This is an example of a performance test case.
-    [self measureBlock:^{
-        // Put the code you want to measure the time of here.
-    }];
-}
+describe(@"changing the count", ^{
+    __block UIView *view;
+    __block RKNotificationHub *subject;
+    
+    before(^{
+        view = [[UIView alloc] initWithFrame:CGRectZero];
+        subject = [[RKNotificationHub alloc] initWithView:nil];
+        [subject setView:view andCount:20];
+    });
+    
+    describe(@"increment", ^{
+        it(@"should increase count", ^{
+            [subject increment];
+            expect([subject count]).to.equal(21);
+        });
+    });
+    
+    describe(@"incrementBy:", ^{
+        it(@"should increase count by given value", ^{
+            [subject incrementBy:10];
+            expect([subject count]).to.equal(30);
```

Comment:
Good question. This is actually a feature of BDD-style syntax: sibling blocks are independent of each other. The order of calls for the first two tests from "changing the count" is like this:
1. before block -> describe "increment" block -> it "should increase count" block
2. before block -> describe "incrementBy:" block -> it "should increase count by given value" block
A new RKNotificationHub object is created with count 20 for each run in the "before" block.